### PR TITLE
[code quality][nacos-address]improve the code of ServiceManager.java

### PR DESCRIPTION
**Pull request by zhongxiongzeng**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by KomachiSion** [comment]
Please fix the cla problem first

**Comment by zhongxiongzeng** [comment]
> Please fix the cla problem first

done

**Comment by KomachiSion** [comment]
Can you describe why you do refactor like this?  I think this change need to be discuss.

**Comment by zhongxiongzeng** [comment]
> Can you describe why you do refactor like this? I think this change need to be discuss.

主要是优化了两个锁，使逻辑简单清晰一点。
1.优化putService线程安全的逻辑：serviceMap是ConcurrentHashMap，本身就是一个线程安全类，加锁-get-put-解锁，用putIfAbsent替代就行了。
2.addUpdatedServiceToQueue这个方法也是一样的，若只要保证这个方法是线程安全的，用synchronized就够了，额外用一个对象来加锁反而让代码显得臃肿复杂

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: naming/src/main/java/com/alibaba/nacos/naming/core/ServiceManager.java [hunk 1]
```diff
@@ -868,14 +859,8 @@ public boolean containService(String namespaceId, String serviceName) {
...
     public void putService(Service service) {
-        if (!serviceMap.containsKey(service.getNamespaceId())) {
-            synchronized (putServiceLock) {
-                if (!serviceMap.containsKey(service.getNamespaceId())) {
-                    serviceMap.put(service.getNamespaceId(), new ConcurrentSkipListMap<>());
-                }
-            }
-        }
-        serviceMap.get(service.getNamespaceId()).putIfAbsent(service.getName(), service);
+        serviceMap.putIfAbsent(service.getNamespaceId(), new ConcurrentSkipListMap<>())
```
[the remark this answers is not in the document]
Comment (zhongxiongzeng) [reply naming/src/main/java/com/alibaba/nacos/naming/core/ServiceManager.java:862]: yes, i wil fix it later

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**